src/AppBundle/Subscriber/AbstractStatusChangeSubscriber.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace AppBundle\Subscriber;
+
+use AppBundle\Issues\Status;
+use AppBundle\Issues\StatusApi;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+abstract class AbstractStatusChangeSubscriber implements EventSubscriberInterface
+{
+    protected static $triggerWordToStatus = [
+        'needs review' => Status::NEEDS_REVIEW,
+        'needs work' => Status::NEEDS_WORK,
+        'works for me' => Status::WORKS_FOR_ME,
+        'reviewed' => Status::REVIEWED,
+    ];
+
+    protected $statusApi;
+
+    public function __construct(StatusApi $statusApi)
+    {
+        $this->statusApi = $statusApi;
+    }
+
+    /**
+     * Parses the text and looks for keywords to see if this should cause any 
+     * status change.
+     *
+     * @param string $body
+     *
+     * @return null|string
+     */
+    protected function parseStatusFromText($body)
+    {
+        $triggerWord = implode('|', array_keys(static::$triggerWordToStatus));
+        $formatting = '[\\s\\*]*';
```

Comment:
FYI you don't have to escape special characters in a character class, except `-` if it's not at the end and to be used as literal.